src/Extensions/UserDefinedFormControllerExtension.php

```diff
@@ -2,95 +2,108 @@
     {
-        // Check form configuration to see if the email should be encrypted
-        if (!$this->owner->encryptEmail()) {
+        // Early exit if this form should not be encrypted
+        if (!$this->owner->isEncryptEmail()) {
             return;
         }
 
         // If the To field is a dynamic field then allow email to be sent without encryption or warning
         $skipEncryption = $recipient->SendEmailToField()->exists();
 
-        $pathToFile = null;
-        $signingCredentials = [];
+        $encryptionFilePath = null;
+        $signingCredentials = null;
 
         if (!$skipEncryption) {
             // Check for a recipient encryption certificate, with a matching email address, and set path to file
             $encryptionCertificateEntry = SmimeEncryptionCertificate::get()
                 ->filter('EmailAddress', $recipient->EmailAddress)->first();
 
             if ($encryptionCertificateEntry && $encryptionCertificateEntry->EncryptionCrt->exists()) {
-                $pathToFile = $this->getFilePath($encryptionCertificateEntry->EncryptionCrt);
+                $encryptionFilePath = $this->getFilePath($encryptionCertificateEntry->EncryptionCrt);
             }
 
             // If no encryption certificate was found then proceed but append a warning to the email.
-            if (!$pathToFile) {
+            if (!$encryptionFilePath) {
                 $subject = $email->getSubject();
                 $encryptionMessage = '[UNENCRYPTED: CHECK CMS CONFIGURATION]';
 
                 $email->setSubject(sprintf('%s %s', $subject, $encryptionMessage));
             }
 
-            $senderEmailAddress = array_key_first($email->getFrom());
+            // Get the Sender Email address
+            $senderEmailAddress = $email->getFrom()[0]->getAddress();
 
+            // Get the credentials for signing an Email
             $signingCredentials = $this->checkForSigningCredentials($senderEmailAddress);
         }
 
-        // Re-registering SmimeMailer for each recipient with appropriate encryption/signing details
-        $this->registerSMIMEMailer(
-            $pathToFile,
-            $signingCredentials
-        );
-    }
-
-    /**
-     * Uses injection to replace the standard Mailer class, used for handling the sending of the email,
-     * with an SMIMEMailer instance with encryption/signature properties.
-     */
-    public function registerSMIMEMailer(?string $pathToEncryptionFile, array $signingCredentials): void
-    {
-        Injector::inst()->registerService(
-            SMIMEMailer::create(
-                $pathToEncryptionFile,
-                $signingCredentials['certificate'] ?? null,
-                $signingCredentials['key'] ?? null,
-                $signingCredentials['passphrase'] ?? null,
-            ),
-            Mailer::class
-        );
+        // Set email signing and encryption properties
+        $email->setSigningCredentials($signingCredentials);
```

Comment:
CORE: Setting the `signing` and `encryption` credentials for the new `SmimeEmail` instance